sdk/eventhubs/azure_messaging_eventhubs/src/common/recoverable_connection.rs

```diff
@@ -822,58 +825,10 @@ impl AmqpReceiverApis for AmqpReceiverClient {
 mod tests {
     use super::*;
     use crate::{consumer, ErrorKind, EventHubsError};
-    use azure_core::credentials::TokenRequestOptions;
-    use azure_core::{credentials::AccessToken, http::Url, Result};
+    use azure_core::http::Url;
     use azure_core_amqp::AmqpError;
-    use std::sync::{Arc, Mutex as SyncMutex};
-    use time::{Duration, OffsetDateTime};
-
-    // Helper struct to mock token credential
-    #[derive(Debug)]
-    struct MockTokenCredential {
-        /// Duration in seconds until the token expires
-        token_duration: i64,
-
-        /// The token itself
-        /// This is a mock token, so we don't need to worry about the actual value
-        token: SyncMutex<AccessToken>,
-    }
-
-    impl MockTokenCredential {
-        fn new(expires_in_seconds: i64) -> Arc<Self> {
-            let expires_on = OffsetDateTime::now_utc() + Duration::seconds(expires_in_seconds);
-            Arc::new(Self {
-                token_duration: expires_in_seconds,
-                token: SyncMutex::new(AccessToken::new(
-                    azure_core::credentials::Secret::new("mock_token"),
-                    expires_on,
-                )),
-            })
-        }
-    }
-
-    #[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
-    #[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
-    impl TokenCredential for MockTokenCredential {
-        async fn get_token(
-            &self,
-            _scopes: &[&str],
-            _options: Option<TokenRequestOptions>,
-        ) -> Result<AccessToken> {
-            // Simulate a token refresh by incrementing the token get count
-            // and updating the token expiration time
-
-            let expires_on = OffsetDateTime::now_utc() + Duration::seconds(self.token_duration);
-            {
-                let mut token = self.token.lock().unwrap();
-                *token = AccessToken::new(
-                    azure_core::credentials::Secret::new("mock_token"),
-                    expires_on,
-                );
-                Ok(token.clone())
-            }
-        }
-    }
+    use azure_core_test::credentials::MockCredential;
```

Comment:
Just to note: if this didn't make sense to use you didn't have to. Just letting you know.

Reply:
Actually it *did* make sense to me. The only reason I didn't use MockCredential before was that I had a need for a strict token timeout implementation (which MockCredential doesn't have). 

But the `RecoverableConnection` tests don't have the same requirements, so it makes sense to use the existing implementation rather than having *three* separate mock credential implementations.